static/js/hotkey.js

```diff
@@ -285,7 +289,7 @@ exports.process_enter_key = function (e) {
         if (emoji_picker.is_composition(e.target)) {
             e.target.click();
         } else {
-            emoji_picker.toggle_selected_emoji();
+            emoji_picker.toggle_selected_emoji(e);
```

Comment:
I think we should do the `shiftKey` check here and just pass a boolean to this function `shift_down` rather than having keydown event format concerns pass through a couple other functions.

Reply:
Or probably better, pass a `{hide_popover: false}` options dictionary that we can extend with the default above.

Reply:
Adding that check in `hotkey.js` doesn't make sense since we also have two other ways to call `emoji_picker.toggle_reaction(emoji_name, event)` (holding shift also works while clicking with the mouse or pressing enter while in the emoji search input). So to implement this indirection without repeating code we would need a `get_toggle_options_from_event(event)` function and call it thrice, at which point I would call this overengineering.

Reply:
Yeah, I like your refactoring; hadn't thought about the mouse case.